### ISSUE TITLE
JWTAuth, custom parseHeadersOrNull

### ISSUE THREAD

**Issue by fraadsbrandth** [ktorio/ktor#1048]
### Ktor Version
1.1.3

### Ktor Engine Used(client or server and name)
netty server

### Feedback
I have an API that sends a JWT as a part of a cookie, not as an Authorization Header.
Then i can't use this feature..

Could it be possible to add a custom way to retrieve the JWT? That of course defaults to the standard.

For instance add a variable function in `JWTAuthenticationProvider `
```
var parseHeadersOrNull : (ApplicationRequest) -> HttpAuthHeader = { it.parseAuthorizationHeaderOrNull() }
```

And then call this function instead of where this is currently called;

```
val token = call.request.parseAuthorizationHeaderOrNull()
```

**Comment by cy6erGn0m** [comment]
@fraadsbrandth what do you think about PR #1050  ?

**Comment by fraadsbrandth** [comment]
looks good @cy6erGn0m

**Comment by cy6erGn0m** [comment]
Fixed in 1.2.0-rc2
